clipdel: add options to delete number of entries

N-R-K opened the pull request:
Adds an `-n` flag which allows deleting certain number of entries from newest (default) or oldest (`-N`).

Allows deleting the current clip with `clipdel -n 1` as requested in https://github.com/cdown/clipmenu/issues/57#issuecomment-366728601 and https://github.com/cdown/clipmenu/issues/98.

Caveat: `-n 1` selects the newest/oldest based on time of entry. But the currently serving clip maybe different if another entry was selected thru `clipmenu`.

Opinion on adding a config option to pull an entry to the top when selecting it to be served via clipmenu? I'd like this personally, since I find it more natural to see the current serving entry on the top when opening clipmenu.

N-R-K committed:
clipdel: refactor into enum match_type

N-R-K committed:
clipdel: add options to delete number of entries

-n will match against the newest entries while -N matches
against oldest ones. can be naturally combined with -v also.

Closes: https://github.com/cdown/clipmenu/issues/98

snprajwal reviewed (commented):
Thank you for this! This will allow us to enable better functionality with passwordstore :)

cdown commented:
(concept looks good, may be a bit to review. thanks for your patience)

cdown commented:
This looks great, thank you. I guess the only thing remaining is an integration test?

N-R-K commented:
> I guess the only thing remaining is an integration test?

I thought I had added them already but apparently not. Added them now. But it seems to be failing the incr test, which seems unrelated (?)

cdown commented:
weird that it was flaky on this and the other pr. let me take a look at what's going on there

(and github, why on earth can't you retry a pr workflow...)

cdown commented:
That actually looks like some real bug rather than a failure, it's failing dupe detection somehow, or some other clip is getting in. I will add more instrumentation to the integration tests to print out the storage so that if it happens again we find out what actually happened.

Anyway, it passed when run again on CI and locally, so merging. Longer term hopefully those will allow us to debug the flakiness or bug. Thank you!

cdown commented:
087cf07022eca27eaa5b69d53fc3809ae7afe801 should hopefully work out what's going on there.
